core/frontend/src/tile/RealityTileTree.ts

```diff
@@ -112,6 +113,104 @@ export interface RealityTileTreeParams extends TileTreeParams {
   readonly rootTile: RealityTileParams;
 }
 
+/** The return status for collecting a tile with [[RealityTileCollector.SelectTile]]
+ * @alpha
+ */
+export enum RealityTileCollectionSelectionStatus { Continue, Reject, Accept }
+
+/**  The return status for collecting a set of tiles with [[RealityTileTree.collectTiles.
+ * @alpha
+ */
+export enum RealityTileCollectionStatus { Loading, Success }
+
+/** @alpha */
+export enum RealityTileDrapeStatus { Loading, Success, Error }
+
+/** Base class for a collecting reality tiles from a [[RealityTileTree]].
+ * @alpha
+ */
+export abstract class RealityTileCollector {
+  public acceptedGeometry(): RealityTileGeometry[] {
+    const geometry = new Array<RealityTileGeometry>();
+    for (const accepted of this.accepted) {
+      if (accepted.geometry)
+        geometry.push(accepted.geometry);
+    }
+
+    return geometry;
+  }
+
+  /** Missing (not yet loaded) tiles.  Loading of these tiles can be requested by calling [[RealityTileCollector.requestMissingTiles]].
+   * @internal
+  */
+  public missing = new Set<RealityTile>();
+  /** The accepted tiles. If [[RealityTileTree.collectTiles]] returns [[RealityTileCollectionStatus.Success]] then this contains complete
+   * collection of loaded tiles accepted by [[RealityTileCollector.selectTile]].
+   * @internal
+   */
+  public accepted = new Array<RealityTile>();

```

Comment:
Is something marking these tiles as in use by a viewport and preventing their content from being disposed?

Reply:
No -- and they really shouldn't be associated to a viewport-- should I make a dummy viewport perhaps?

Reply:
We determine whether or not a tile is in use by seeing if at least one Viewport is using it. Ideally these tiles would be recorded as "in use" by any viewports that are using them to produce drapes.

Reply:
These tiles should be disposed as soon as the collection is successful and the geometry is extracted from them.  They really are not associated to a viewport - but there isn't a way to request without one.

Reply:
Resolved by `TileUser`.